x-pack/plugin/repository-encrypted/src/internalClusterTest/java/org/elasticsearch/repositories/encrypted/EncryptedAzureBlobStoreRepositoryIntegTests.java

```diff
@@ -76,6 +89,17 @@ protected String repositoryType() {
         return EncryptedRepositoryPlugin.REPOSITORY_TYPE_NAME;
     }
 
+    @Override
+    protected BlobStore newBlobStore(BlobStoreRepository blobStoreRepository) {
+        EncryptedRepository.EncryptedBlobStore blobStore = (EncryptedRepository.EncryptedBlobStore) super.newBlobStore(blobStoreRepository);
+        if (false == blobStoreRepository.isReadOnly()) {
+            PlainActionFuture.get(
+                l -> threadPool.generic().execute(ActionRunnable.run(l, () -> blobStore.maybeInitializePasswordGeneration()))
+            );
+        }
+        return blobStore;
+    }
```

Comment:
Some integration tests directly access the blob store for writes.
But this bypasses the repository initialization (ie creating the initial password generation) which kicks in when a snapshot is started, which regularly happens before a blob write.
This artificially introduces the same initialization so the blobStore can be used for writes.